src/components/auth/AuthWrapper.tsx

```diff
@@ -1,23 +1,67 @@
 import { Configuration, PublicClientApplication } from "@azure/msal-browser";
-import { MsalProvider } from "@azure/msal-react";
+import { MsalProvider, useIsAuthenticated, useMsal } from "@azure/msal-react";
 import { applicationConfig } from "config";
-import React, { FunctionComponent } from "react";
+import React, { createContext, FunctionComponent } from "react";
 
-const authorityBaseUrl = "https://login.microsoftonline.com";
+export const AuthWrapper: FunctionComponent = ({ children }) => {
+  return (
+    <MsalProvider instance={pca}>
+      <MsalShim>{children}</MsalShim>
+    </MsalProvider>
+  );
+};
+
+const MsalShim: FunctionComponent = ({ children }) => {
+  /**
+   * Wrapper for the MSAL auth context.
+   *
+   * @remarks
+   * Acts as a shim between MSAL and the Beacons Backoffice application so that high-level components can consume
+   * authenticated user data without depending on a concrete auth provider.
+   *
+   */
+  const currentUser = useMsal().instance.getAllAccounts()[0] || {};
```

Comment:
Not 💯  but thinking that, you could just use pca here like this:
```suggestion
  const currentUser = pca.getAllAccounts()[0] || {};
```

beacuse we did something similar in the authGateway to get the current user: https://github.com/mcagov/beacons-backoffice/blob/cf9896bd11432ae8c7dddff115001f50b5bb3884/src/gateways/AuthGateway.tsx#L14

Reply:
Great thinking!  I had it this way initially, but the React component this code appears doesn't update the context without use of the `useMsal()` hook.  So it was working like this:

1.  Unauthenticated user first goes to webpage.  PCA is instantiated.
2. `<AuthWrapper>` renders and provides auth data to child components via React.context.  Because user is not yet authenticated, auth 'stuff' is empty.
3.  User authenticates.
4. Context provided by `<AuthWrapper>` is still empty because nothing has prompted it to re-render and update its values from `pca`.
5. So the user would log in but the `<UserMenu>` component would still think there was no authenticated user.

Using the `useMsal()` hook ([docs](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsal-hook)) means the shim'd context `<AuthContext>` provides downwards is updated from the MSAL context that we don't want to expose to our application's internals.  So now:

1 - 3 as above.
4. Context provided by `useMsal()` is updated, which prompts an update to our shim context provided by `<AuthWrapper>`, which becomes available to context consumers like `<UserMenu>`.
5. `<UserMenu>` can play back the authenticated user.

So whenever the user's authentication state changes `<AuthWrapper>` provides the most up-to-date context downwards.  I hope that makes sense, it took me a while to work out why using just `pca` wasn't updating after the user authenticated... 😬

Reply:
Ohh okay, that helps a lot, thank you for explaining it so well!! I can definitely imagine that not being very easy to figure out 😅